flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/serde/DynamicTableSinkSpecSerdeTest.java

```diff
@@ -139,39 +139,166 @@ public void testDynamicTableSinkSpecSerde() throws IOException {
                                             }
                                         })));
-        spec2.setReadableConfig(new Configuration());
 
-        Map<String, String> properties3 = new HashMap<>();
-        properties3.put("connector", "values");
-        properties3.put("schema.0.name", "a");
-        properties3.put("schema.0.data-type", "BIGINT");
-        properties3.put("schema.1.name", "b");
-        properties3.put("schema.1.data-type", "INT");
-        properties3.put("schema.2.name", "m");
-        properties3.put("schema.2.data-type", "STRING");
-        properties3.put("writable-metadata", "m:STRING");
-
-        final CatalogTable catalogTable3 = CatalogTable.fromProperties(properties3);
+        Map<String, String> options3 = new HashMap<>();
+        options3.put("connector", TestValuesTableFactory.IDENTIFIER);
+        options3.put("writable-metadata", "m:STRING");
 
         final ResolvedSchema resolvedSchema3 =
                 new ResolvedSchema(
                         Arrays.asList(
                                 Column.physical("a", DataTypes.BIGINT()),
                                 Column.physical("b", DataTypes.INT()),
-                                Column.physical("m", DataTypes.STRING())),
+                                Column.metadata("m", DataTypes.STRING(), null, false)),
                         Collections.emptyList(),
                         null);
+        final CatalogTable catalogTable3 =
+                CatalogTable.of(
+                        Schema.newBuilder().fromResolvedSchema(resolvedSchema3).build(),
+                        null,
+                        Collections.emptyList(),
+                        options3);
 
         DynamicTableSinkSpec spec3 =
                 new DynamicTableSinkSpec(
-                        ObjectIdentifier.of("default_catalog", "default_db", "MyTable"),
-                        new ResolvedCatalogTable(catalogTable3, resolvedSchema3),
+                        ContextResolvedTable.temporary(
+                                ObjectIdentifier.of(
+                                        DEFAULT_BUILTIN_CATALOG,
+                                        DEFAULT_BUILTIN_DATABASE,
+                                        "MyTable"),
+                                new ResolvedCatalogTable(catalogTable3, resolvedSchema3)),
                         Collections.singletonList(
                                 new WritingMetadataSpec(
                                         Collections.singletonList("m"),
                                         RowType.of(new BigIntType(), new IntType()))));
-        spec3.setReadableConfig(new Configuration());
 
-        return Arrays.asList(spec1, spec2, spec3);
+        return Stream.of(spec1, spec2, spec3);
+    }
+
+    @ParameterizedTest
+    @MethodSource("testDynamicTableSinkSpecSerde")
+    void testDynamicTableSinkSpecSerde(DynamicTableSinkSpec spec) throws IOException {
+        TableEnvironmentImpl tableEnv =
```

Comment:
use `PlannerMocks` instead? It is not perfect yet, but you can expose the `Configuration` (not `TableConfig`) and should be able to access everything you need.